UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/ApplierImpl.cs

```diff
@@ -187,7 +193,7 @@ public void Apply(TensorProxy tensorProxy, IEnumerable<Agent> agents)
 
             foreach (var agent in agents)
             {
-                var memory = agent.GetMemoriesAction();
+                var memory = m_Memories.ContainsKey(agent.Info.id) ? m_Memories[agent.Info.id] : null;
```

Comment:
@surfnerd is there a better way to do this (in one lookup)? We were trying m_Memories.ElementAtOrDefault(id).Value but that wasn't behaving as expected.

Reply:
`ElementAtOrDefault` takes an index as a parameter.  That probably explains the unexpected behavior.  You can always use `TryGetValue` which will return the `default` value if it is not in the `Dictionary`.

Reply:
🤦‍♂  So 
```
List<float> memories = null;
m_Memories.TryGetValue(agent.Info.id, memories);
```
?

Reply:
yeah, you could combine it with the if statement down below as well. 

```
List<float> memories = null;
if (m_Memories.TryGetValue(agent.Info.id, out memories) || memory.Count < memorySize * m_MemoriesCount)
{
    ...
}
...
```